content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+Using the @presenceChannel@ we mark the presence of the user and also subscribe to @enter@. We also publish the @userInfo@ data to the @chatChannel@.
+Then using the @chatChannel@ we subscribe to the @userInfo@ and get user data.
+Lastly using the @chatChannel@ we subscribe to the @chatMessage@ to get the updates coming in to the newsfeed. We then update the state variables with the new update.
+
+Now, add the following imports statements on top of the file after the @Ably@ import statement to enable use of semantic-ui:
+
+```[javascript]
+import { Container, Header, Divider, Button, Form, TextArea } from 'semantic-ui-react'
+```
+
+Now, finally let’s add the @sendMyMessage@ function to the @App@ component and the HTML to post an update:
+
+```[javascript]
+function sendMyMessage() {
+        const newMsgId = "msg-id-" + Math.random().toString(36).substr(2, 6);
+        setTodoInput("");
+        if (todoInput !== "") {
+            chatChannel.publish("chatMessage", {
+                message: todoInput,
+                localMsgId: newMsgId,
+                date: new Date()
+            });
+            setState(prevState => {
+                return {
+                  ...prevState,
+                  msgs: [...prevState.msgs, {"summary": todoInput, "image": my.avatar, "date": new Date(), "name": my.name}].sort((a,b)=>b.date-a.date)
+                };
+            });
+        }
+    }
+
+    return (
+        <Container textAlign='left' className='App-header'>
+          <Header as='h2'>Ably React Tutorial - RealTime</Header>  
+          <Form>
+             <TextArea 
+                value={todoInput}
+                placeholder="Post an update"
+                onChange={e => setTodoInput(e.target.value)}
+             />
+             <div className="App-button"><Button content='Send' primary onClick={sendMyMessage} /></div>
+         </Form>
+             
+         <Divider />
+         
+         </Container>
+    )
+```
+
+As you can see in the @sendMyMessage@ function we publish the message to @chatMessage@ using @chatChannel@ and update the local state accordingly with the new update posted by the user.
+
+"See this step in Github":https://github.com/ably/tutorials
+
+h2. Step 4 - Displaying the updates in the newsfeed
+Firstly, update the following imports statements in @App.js@ file:
+
+```[javascript]
+import React, { useState, useEffect, useRef } from "react";
+import {Container, Header, Divider, Label, Button, Form, TextArea } from "semantic-ui-react";
+import UICard from "./UICard";
+```
+
+Now, add the following @scrollRef@ to the reset the scroll, on the top of the @App@ component above the @todoInput@ state:
+
+```[javascript]
+const scrollRef = useRef(null);
+const [todoInput, setTodoInput] = useState("");
+```
+
+Add the function @getMessages@ below the @sendMyMessage@ function to update the newsfeed with the new updates:
+
```

Comment:
As with the rest of the comments, all these code snippets need more explanation, please.